Userland/Services/WindowServer/Overlays.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/IntrusiveList.h>
+#include <AK/Vector.h>
+#include <LibGfx/Painter.h>
+#include <WindowServer/MultiScaleBitmaps.h>
+#include <WindowServer/Screen.h>
+
+namespace WindowServer {
+
+class Screen;
+
+class Overlay {
+    friend class Compositor;
+
+public:
+    virtual ~Overlay();
+
+    enum class ZOrder {
+        ScreenNumber,
+    };
+    virtual ZOrder zorder() const = 0;
```

Comment:
```suggestion
    [[nodiscard]] virtual ZOrder zorder() const = 0;
```

This would make misusing the getter a little harder. Same for the other getters.